### PR TITLE
Milestone 6: Feature/friendshipv2

### PR DESCRIPTION
**Pull request by dansantander**
This milestone complies with the following requirements:
- Try to implement friendships with 2 rows for mutual friendship.
- Add tests for this feature.

### CODE REVIEW

File: app/controllers/friendships_controller.rb [hunk 1]
```diff
@@ -33,8 +33,10 @@ def update
     user_id = current_user.id
     @friendship = Friendship.find_by(confirmer: user_id)
     accepted = @friendship.update_attribute(:confirmed, params[:confirmed] = true)
+    # requester_id = @friendship.requester.id
 
     if accepted
+      # Friendship.new(confirmer_id: requester_id, requester_id: user_id, confirmed: true).save
       redirect_to user_path(user_id), notice: 'Friend request Succesfully Accepted'
```
Comment (uryelah) [inline app/controllers/friendships_controller.rb]: - [x] For some reason the inverted friendhip related code is commented out. Make sure you uncomment or delete unused code before asking for a new review.

File: app/models/friendship.rb [hunk 2]
```diff
@@ -1,10 +1,10 @@
-class DuplicateValidator < ActiveModel::Validator
-  def validate(record)
-    friend1 = Friendship.find_by(requester_id: record.requester_id, confirmer_id: record.confirmer_id)
-    friend2 = Friendship.find_by(requester_id: record.confirmer_id, confirmer_id: record.requester_id)
-    record.errors[:base] << 'This relationship already exists' unless friend1.nil? && friend2.nil?
-  end
-end
+# class DuplicateValidator < ActiveModel::Validator
```
Comment (uryelah) [inline app/models/friendship.rb]: - Please remove commented out code.

File: app/models/friendship.rb [hunk 3]
```diff
@@ -16,10 +16,17 @@ class Friendship < ApplicationRecord
   belongs_to :confirmer, foreign_key: :confirmer_id, class_name: 'User'
   belongs_to :requester, foreign_key: :requester_id, class_name: 'User'
 
-  # belongs_to :confirmer, :class_name => "User"
-  # belongs_to :requester, :class_name => "User"
-
   validates :confirmer, uniqueness: { scope: :requester, message: 'Friend Request Already Sent' }
-  validates_with DuplicateValidator, on: :create
+  # validates_with DuplicateValidator, on: :create
```
Comment (uryelah) [inline app/models/friendship.rb]: -  Please remove commented out code.

**Review by uryelah — CHANGES_REQUESTED** [review]
Hi **Daniel Cruz Santander & Alfred Ezaka**, I'm Sarah your reviewer for this PR 👩‍🚀 

Good job with your duplicated Friendship row, now just make sure the code is not commented out and it's well organized and you'll be good to go! 💪 

# Required Changes 🏗️ 🐛 💡

- [x] Remove unused code like the line 
`@i = Friendship.find_by(confirmer: current_user.id)`
at the friendships controller.

- [x] Try to name your variables with meaningful names. Avoid names constituted of single letters and write them to describe what they contain.

**Comment by uryelah** [comment]
-----

 **Kindly fix the pointed out issues before requesting the next code review**

------------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Review by uche-inyama — APPROVED** [review]
## Approved

Hi team :wave: 

The changes you made are approved.

Go on with the next milestone.

#### ``` Have a great day ``` :slightly_smiling_face: 

``` Uchechukwu Henry Inyama ```

**Pull request merged**